### DISCUSSION TITLE
ERROR, ITMS-90023: Missing required icon file

### DISCUSSION THREAD

**Discussion by mptsuei** [airsdk/Adobe-Runtime-Support#3162] [General]
Hi
I'm uploading my APP to App Store but receives the following error, Please help to sovle the problems.
I am using the Adobe Animate to develop the APP. I have struggleing for may days. Thanks in advance.

ITMS-90023: Missing required icon file - The bundle does not contain an app icon for iPad of exactly '167x167' pixels, in .png format for iOS versions supporting iPad Pro. To support older operating systems, the icon may be required in the bundle outside of an asset catalog. Make sure the Info.plist file includes appropriate entries referencing the file. See https://developer.apple.com/documentation/bundleresources/information_property_list/user_interface

ITMS-90023: Missing required icon file - The bundle does not contain an app icon for iPad of exactly '152x152' pixels, in .png format for iOS versions >= 10.0. To support older operating systems, the icon may be required in the bundle outside of an asset catalog. Make sure the Info.plist file includes appropriate entries referencing the file. See https://developer.apple.com/documentation/bundleresources/information_property_list/user_interface

ITMS-90704: Missing app icon - Include a large app icon as a 1024 by 1024 pixel PNG in the asset catalog of apps built for iOS, iPadOS, or watchOS. Without this icon, apps can't be submitted for review. For details, visit: https://developer.apple.com/documentation/xcode/managing-assets-with-asset-catalogs.

I am sure that XＭL include all icons images. Thanks a lot.

**Comment by FliplineStudios** [comment]
Which version of the AIR SDK are you using, and is this on Mac or PC?  Also I'd recommend unzipping the IPA file to see if those items are missing, to see whether the icon PNGs are packaged and if they're included in the Info.plist file:  For the first two, those should show up as "Icon-167.png" and "Icon-Ipad-152.png", and they would be listed in the Info.plist under CFBundleIconFiles.  For the last item, that would be added to your Assets.car that you generated from Xcode or some other tool.

**Reply by mptsuei** [reply]
Thanks for your help. I am using the AIR SDK 50.2.4.5 for Windows. After unzipping the ipa file, the "Icon-167.png" and "Icon-Ipad-152.png" are in the package. And, they are also listed in the Info.plist under CFBundleIconFile.  Can you tell me how to check whether the  1024 by 1024 pixel PNG file is listed in the Assets.car? Thanks for your help.

![Snap22](https://github.com/airsdk/Adobe-Runtime-Support/assets/102018852/21417d7a-68f4-4e34-b45b-33211a5886c6)

**Reply by FliplineStudios** [reply]
Hmm strange, I'm not sure how to check the Assets.car file... there have been other issues with Windows builds where they're flagged by Apple's systems, I thought those were mostly resolved but I wonder if this is another situation where the LLVM builds from Windows are getting flagged now.

Do you have access to a Mac to package this, while also setting `-platformsdk` to point to the iOS SDK included in Xcode?  This should at least get around any LLVM-specific issues that are being flagged for some reason by Apple, curious if that would stop these new errors too.  Or maybe there's something going wrong on Apple's end right now with submissions...

**Reply by ajwfrost** [reply]
With a mac you could definitely check the contents of the .car file to ensure that has the required size images in it. There's a Rust project that could give a cross-platform way of checking, at https://github.com/vaguilar/carutil but the simplest option would be to use the Distriqt Assets.car generation script..

What's your minimum iOS version set to? This may affect the requirements.

It's a bit odd that it's saying "The bundle does not contain an app icon for iPad of exactly '167x167' pixels, in .png format" if you have checked that the `CFBundleIconFiles` entry is there, and the file exists (and I assume you've checked that it is exactly 167x167 .. although I'm 99% sure ADT will check this too). So potentially those ones are also just missing from the Assets.car file..?

**Comment by marchbold** [comment]
Likely something is missing from your `Assets.car`. Have you tried using the Assets catalogue generation script I created? 

https://github.com/distriqt/AIR-ImageScripts

**Comment by mrfrasier** [comment]
I ran into this issue myself while using an Assets.car generated with the latest Xcode. By default, the AppIcon only asks for 1024px size. If you open the settings on the right side and change iOS to "All Sizes", then you'll see a bunch more sizes to add. This fixed my upload issue.

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/4335972/9bdb27ab-3f1c-465e-988c-60aff82ff33d)
